### ISSUE TITLE
Scripts in /etc/rsnapshot/{hosts} does not get regenerated 

### ISSUE THREAD

**Issue by Gomez** [debops/ansible-rsnapshot#8]
I have rsnapshot with debops servers and some external server running. There was a error with the my fqdn so the mail info in the runner script was wrong. I deleted all /etc/rsnapshot/{hosts} directories, but the does not get regenerated.

Small bug, but worth to get documented.

**Comment by drybjed** [comment]
Weird... I'll look into it. Thanks.

**Comment by Gomez** [comment]
Oh, external servers where generated, but not the debops servers.

**Comment by drybjed** [comment]
BTW, do you have backups that take longer than ~24h? I'm working on a solution that handles these cases but I have problem with monthly backups... They'll probably be working, I'll check in a week (August).

**Comment by Gomez** [comment]
I have servers with 1TB space to backup. Dont know how log it takes. Can test it...

**Comment by Gomez** [comment]
Ok, this is still a issue. The debops server got skipped:

```
TASK [debops.rsnapshot : Create server configuration directories on client] ****
skipping: [hershel] => (item=glenn) 
skipping: [hershel] => (item=michonne) 
skipping: [hershel] => (item=dale) 
skipping: [hershel] => (item=merle) 
```

Problem is ansible_fqdn in this task:
```
TASK [debops.rsnapshot : debug] ************************************************
fatal: [hershel]: FAILED! => {"failed": true, "msg": "ERROR! 'dict object' has no attribute 'ansible_fqdn'"}
```
But:

```
[master]$ ansible -m setup merle | grep fqdn
[WARNING]: log file at ansible.log is not writeable, aborting

        "ansible_fqdn": "merle.portknox.de", 
```

What could be wrong here?

**Comment by drybjed** [comment]
Is that host in either `[debops_service_rsnapshot]` or `[debops_service_rsnapshot_rsync]` groups?

**Comment by Gomez** [comment]
Yes, external server works without a problem.

**Comment by drybjed** [comment]
Is that on Ansible v2? I think I know what's happening... I had to [set the FQDN fact specifically](https://github.com/debops/ansible-pki/blob/master/tasks/main.yml#L3-L5) in the play for other hosts to see it. I guess `debops.rsnapshot` could use the same, and the FQDN could be configurable using a default variable as well. I'll check it when I have some time.

**Comment by Gomez** [comment]
Yes Ansible v2. Thanks for explanation!

**Comment by Gomez** [comment]
This bite me again and i tried to fix but failed.

Here is my PR:
https://github.com/Gomez/ansible-rsnapshot/tree/fix_8

I am not sure how to handle the change to this task:
https://github.com/debops/ansible-rsnapshot/blob/162894dc29a3ac77815f645981d2d9c975fd1798/tasks/main.yml#L68
@drybjed Would you take a look, please?

**Comment by drybjed** [comment]
@Gomez Yes, that seems to be a good direction to fix this. I had to use similar approach in a different role as well. Check if the change I suggested works.

**Comment by Gomez** [comment]
Ok, to get the rsnapshot files  correctly generated (with the current version of this role) i need to set ansible_fqdn = "xx.host.de" in the hosts file. 

Right now i only have ansible_ssh_host set. Is this a change in Ansible (i use 2.1.1.0)? I assumed ansible_fqdn is set by Ansible itself. 

For some hosts i have set fqdn = "xx.host.de", this was for bootstrapping. @drybjed Is this still used?
